### PR TITLE
chore(poseidon): add verification key json files

### PR DESCRIPTION
**Pull request by sripwoud**
Generated with:
1. create `options.json`:
   ```json
   {
      "path/to/zk-kit.circom/packages/poseidon-proof/circomkit.json": {
          "circuit": "poseidon-proof",
          "paramsList": [["1"],["2"],["3"],["4"],["5"],["6"],["7"],["8"],["9"],["10"],["11"],["12"],["13"],["14"],["15"],["16"]]
       }
    }
   ```
2. 

    ```commandline
    pnpm add -g @zk-kit/artifacts-cli
    cd $(mktemp -d)
    snarkli gb path/to/options.json $(pwd)
    circuit=poseidon
    for i in {1..16};do mv -f "$circuit-proof-$i/$circuit-proof/groth16_vkey.json" "/path/to/snark-artifacts/packages/$circuit/circuit-$i.json";done
    for i in {1..16};do mv -f "$circuit-proof-$i/$circuit-proof/groth16_pkey.zkey" "/path/to/snark-artifacts/packages/$circuit/$circuit-$i.zkey";done
    for i in {1..16};do mv -f "$circuit-proof-$i/$circuit-proof/$circuit-proof_js/$circuit-proof.wasm" "/path/to/snark-artifacts/packages/$circuit/$circuit-$i.wasm";done
    ```

## Test
See https://github.com/privacy-scaling-explorations/snark-artifacts/pull/143
```commandline
git checkout test/poseidon-verifkey
pnpm --filter ./packages/poseidon t
```

### CODE REVIEW

File: packages/poseidon/package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "@zk-kit/poseidon-artifacts",
   "description": "zk-kit poseidon artifacts",
   "license": "MIT",
-  "version": "1.0.0-beta.1",
+  "version": "1.0.0-beta.4",
```
Comment (sripwoud) [inline packages/poseidon/package.json:5]: to trigger ci and match https://github.com/privacy-scaling-explorations/zk-kit/blob/687bddcb2a124b478af278729f44dc7632eb34d6/packages/poseidon-proof/package.json#L3

**Review by cedoor — APPROVED** [review]
Should we test them before merging?

**Comment by sripwoud** [comment]
Would be better
Yes
But I wasn't sure what was the easiest way.
A jest test, using snakjs CLI directly?

**Comment by cedoor** [comment]
> Would be better Yes But I wasn't sure what was the easiest way. A jest test, using snakjs CLI directly?

Do we need the CLI? All the artifacts are ready to be used. So snarkjs can be used to generate the proof and then to verify it right?

**Comment by sripwoud** [comment]
@cedoor I added tests that cover all the `wasm`, `zkey` , `json` artifacts...
and they are all failing.

Either my test suite itself is wrong (although it is very similar to what we do [here](https://github.com/privacy-scaling-explorations/zk-kit/blob/main/packages/poseidon-proof/tests/index.test.ts))
or all the verification keys I generated are wrong...

**Comment by sripwoud** [comment]
@cedoor New artifacts generated and tests added in https://github.com/privacy-scaling-explorations/snark-artifacts/pull/143: should we merge #143 btw for reference (or even add it into the CI)?

**Comment by sripwoud** [comment]
Reason why my initial tests were wrong was because I was using incompatible zkey and vkey artifacts.  
If 2 different setups were run for a same circuit, producing `{v,z}key{A,B}`,  
one cannot 
- generate a proof with `zkeyA`
- and then verify it with `vkeyB`

This will fail (which is what i was initially doing)

**Comment by cedoor** [comment]
> @cedoor New artifacts generated and tests added in #143: should we merge #143 btw for reference (or even add it into the CI)?

Yea, I'd add it to the CI 👍🏽

**Comment by sripwoud** [comment]
@cedoor branch is rebased
#144 refactors the tests (the ci already picks them)

**Comment by cedoor** [comment]
@sripwoud kk, then this PR looks ok to me!

**Pull request merged**